### DISCUSSION TITLE
Check If Path Exists In XML

### DISCUSSION THREAD

**Discussion by JJWest12** [nextgenhealthcare/connect#5495] [Q&A] [answered]
I am trying to write a basic transformer that determines if a CCD has a providerOrganization tag in it. I have tried using the "!" and not using it and I keep getting mixed results. What is the best way of determining if an XML path exists in code?

    if(!msg['recordTarget']['patientRole']['providerOrganization']){
        return true
    }
    else{
        return false
    }

**Comment by tonygermano** [comment, marked as answer]
If `msg` is an xml object, then `msg['recordTarget']['patientRole']['providerOrganization']` will always return an xml object (an XMLList, actually) which will always be truthy. You can check the length of the list to see if any nodes exist at that path or not.
```js
if(msg['recordTarget']['patientRole']['providerOrganization'].length() > 0){
    return true
}
else{
    return false
}

**Reply by pacmano1** [reply]
@tonygermano has examples, the forums has examples, google has examples.  Mirth uses e4x, see https://deviltechie.wordpress.com/2012/04/25/check-if-an-xml-node-or-attribute-exists/

**Reply by agermano** [reply]
Wrong Germano. Good luck tho.

**Reply by pacmano1** [reply]
Sorry @agermano again!   @tonygermano is that user name on the forums and Mirth slack, bad habit of mine.
